protocol/x/subaccounts/keeper/negative_tnc_subaccount.go

```diff
@@ -37,16 +44,20 @@ func (k Keeper) getNegativeTncSubaccountSeenAtBlock(
 }
 
 // SetNegativeTncSubaccountSeenAtBlock sets a block number in state where a negative TNC subaccount
-// was seen. This function will overwrite previous values at this key.
+// was seen for a specific collateral pool. This function will overwrite previous values at this key.
 // This function will panic if the old block height is greater than the new block height.
 func (k Keeper) SetNegativeTncSubaccountSeenAtBlock(
 	ctx sdk.Context,
+	collateralPoolAddr sdk.AccAddress,
```

Comment:
So collateral pools are tied to `sdk.AccAddress`, i.e Subaccounts module? shouldn't it be tied to an isolated market?

Reply:
Ok, talked to jay, seems like if we use the subaccounts module string that's the default cross margined iso pool.
Can we add this as a comment to make it more clear? also function above.

Reply:
Actually, i see [here](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/subaccounts/keeper/subaccount.go#L83) we are making an account address based on `"subaccount:<perpId>"`. Is there a reason we are using an sdk.Address`sdk.AccAddress` as the type of the key? Generating an acc address will hash the information so it's unreadable. I think collat pools are based off of perp ids purely. Would it make more sense to seperate the concept of an account address and use purely a `<prefix>:<perp id>` for this?

Reply:
seems like the only possible params here are <perp id> or <default cross margin perp id> so not sure if we should shoehorn this in an AccAddress

Reply:
Discussed in slack, the actual place collateral is exiting / entering are sdk.AccAddress which can be derived from a perpetual Id. However, these accounts are associated with the `x/subaccounts` module so I don't think the addresses should be disassociated from the `x/subaccounts` module by removing the prefix.